File: articles/numerical-methods-and-algorithmic-approximation/java/NumericalApproximation.java
public class NumericalApproximation {
  static double f(double x) { return Math.sin(x) + 0.25 * x * x; }
  static double centralDifference(double x, double h) { return (f(x + h) - f(x - h)) / (2.0 * h); }
  public static void main(String[] args) { System.out.printf("%.12f%n", centralDifference(1.0, 0.01)); }
}

File: articles/numerical-methods-and-algorithmic-approximation/go/numerical_approximation.go
package main

import (
  "fmt"
  "math"
)

func f(x float64) float64 { return math.Sin(x) + 0.25*x*x }
func centralDifference(x, h float64) float64 { return (f(x+h) - f(x-h)) / (2*h) }
func main() { fmt.Printf("%.12f\n", centralDifference(1.0, 0.01)) }
